doc/api/crypto.md

```diff
@@ -1813,28 +1813,35 @@ must be an object with the properties described above.
 <!-- YAML
 added: v11.6.0
 changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/???
+    description: The `key` argument can now be a `KeyObject` with type
+                 `private`.
   - version: v11.7.0
     pr-url: https://github.com/nodejs/node/pull/25217
     description: The `key` argument can now be a private key.
 -->
-* `key` {Object | string | Buffer}
+* `key` {Object | string | Buffer | KeyObject}
   - `key`: {string | Buffer}
   - `format`: {string} Must be `'pem'` or `'der'`. **Default:** `'pem'`.
   - `type`: {string} Must be `'pkcs1'` or `'spki'`. This option is required
     only if the `format` is `'der'`.
 * Returns: {KeyObject}
 
 Creates and returns a new key object containing a public key. If `key` is a
-string or `Buffer`, `format` is assumed to be `'pem'`; otherwise, `key`
-must be an object with the properties described above.
+string or `Buffer`, `format` is assumed to be `'pem'`; if `key` is a `KeyObject`
+with type `'private'`, the public key is derived from the given private key;
+otherwise, `key` must be an object with the properties described above.
 
 If the format is `'pem'`, the `'key'` may also be an X.509 certificate.
 
 Because public keys can be derived from private keys, a private key may be
 passed instead of a public key. In that case, this function behaves as if
 [`crypto.createPrivateKey()`][] had been called, except that the type of the
-returned `KeyObject` will be `public` and that the private key cannot be
-extracted from the returned `KeyObject`.
+returned `KeyObject` will be `'public'` and that the private key cannot be
+extracted from the returned `KeyObject`. Similarly, if a `KeyObject` with type
+`'private'` is given, a new `KeyObject` with type `'public'` will be returned
```

Comment:
nit: I would rephrase  to remove "private keys" and "KeyObjects of type 'private'" being described seperately, and emphasize that private keys cannot be derived from public key objects, ever (regardless of how they are created).

Suggestion:

> Because public keys can be computed from private keys, a private key or `KeyObject` of type `'private'` may be passed instead of a public key. When used this way, the argument types are the same as￼ [`crypto.createPrivateKey()`][], but a `KeyObject` of type `'public'` is returned. Note that the inverse is never possible, a private key *cannot* be derived from a public key or `KeyObject`.

Reply:
I like your description, but since private key objects cannot be passed to `crypto.createPrivateKey` (yet), I am afraid it might be incorrect (but maybe that's just my interpretation of it).